File: v1/src/main/java/com/api/v1/mappers/CreditCardResponseMapper.java
package com.api.v1.mappers;

import com.api.v1.domain.CreditCard;
import com.api.v1.dtos.CreditCardResponseDto;

public record CreditCardResponseMapper(CreditCard creditCard) {

    public CreditCardResponseDto mapToDto() {
        return new CreditCardResponseDto(
                creditCard.getOwnerName(),
                creditCard.getCardNumber(),
                creditCard.getDueDate()
        );
    }

}

File: v2/src/main/kotlin/com/api/v2/creditcard/mappers/CreditCardResponseMapper.kt
package com.api.v2.creditcard.mappers

import com.api.v2.creditcard.domain.CreditCard
import com.api.v2.creditcard.dtos.CreditCardResponseDto

class CreditCardResponseMapper {

    companion object {
        fun mapToDto(creditCard: CreditCard): CreditCardResponseDto {
            return CreditCardResponseDto(
                creditCard.cardNumber,
                creditCard.flag,
                creditCard.dueDate
            )
        }
    }

}